ui/src/components/CubePreviewValue.vue

```diff
@@ -4,7 +4,7 @@
   </span>
   <router-link v-else-if="isResource" :to="{ name: 'ResourcePreview', params: { resourceId: value.id.value, sharedTerm: isSharedTerm } }" class="tag is-rounded is-small">
     <term-with-language :values="label" :selected-language="selectedLanguage">
-      <term-display :term="value.id" :base="cubeUri" />
+      <external-term :resource="value" :base="cubeUri" />
```

Comment:
This is a bit weird.

Here `term-label` is used as a fallback for when `label` has no value for the provided language. `external-term` should probably replace `term-with-language` in this case.

Reply:
Ok, I think I gotcha. Moved the label handling to `<external-term>` and also compose the `term-with-language`